model/annotations/task_annotations.go

```diff
@@ -207,6 +207,62 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 	return errors.Wrapf(err, "adding task annotation for task '%s'", a.TaskId)
 }
 
+// PatchAnnotation adds issues onto existing annotations.
+func PatchAnnotation(a *TaskAnnotation, userDisplayName string, upsert bool) error {
+	existingAnnotation, err := FindOneByTaskIdAndExecution(a.TaskId, a.TaskExecution)
+	if err != nil {
+		return errors.Wrapf(err, "finding task annotation")
+	}
+	if existingAnnotation == nil && upsert {
+		if !upsert {
+			return errors.Errorf("annotation for task '%s' and execution %d not found", a.TaskId, a.TaskExecution)
+		} else {
+			return UpdateAnnotation(a, userDisplayName)
+		}
+	}
+
+	if a.Note != nil {
```

Comment:
how about if no annotation exists? do we want to allow a note or metadata then?

Reply:
that's for `existingAnnotation == nil && upsert`, but how about for `existingAnnotation == nil && !upsert`?